ssl/t1_lib.c

```diff
@@ -2379,6 +2387,15 @@ int tls_choose_sigalg(SSL *s, int *al)
                     } else if (lu->sig_idx != s->cert->key - s->cert->pkeys) {
                             continue;
                     }
+                    if (lu->sig == EVP_PKEY_RSA_PSS) {
+                        /* validate that key is large enough for the signature algorithm */
+                        const RSA *rsa = EVP_PKEY_get0_RSA(s->cert->pkeys[SSL_PKEY_RSA_PSS_SIGN].privatekey);
```

Comment:
and, of course, same comment as above.  Maybe even make a function that takes rsa and hash as parameters.